proxy/request_test.go

```diff
@@ -0,0 +1,25 @@
+package proxy
+
+import (
+	"testing"
+)
+
+func TestBadMessage(t *testing.T) {
+	req := ""
+	resp := handleRequest([]byte(req))
+
+	expected := `{"id":null,"error":{"errcode":"M_NOT_JSON","error":"unexpected end of JSON input"}}`
```

Comment:
Asserting on the language-provided error text is a little strange; I would probably just assert on errcode and non-empty error